src/SonarQube.Client.Tests/Api/SonarQubeService_GetAllOrganizationsAsync.cs

```diff
@@ -0,0 +1,192 @@
+      ""url"": ""https://www.bar.com"",
+      ""avatar"": ""https://www.bar.com/logo.png"",
+      ""guarded"": false
+    }
+  ]
+}");
+
+            var result = await service.GetAllOrganizationsAsync(CancellationToken.None);
+
+            messageHandler.VerifyAll();
+
+            result.Should().HaveCount(2);
+            result.Select(x => x.Key).Should().BeEquivalentTo(new[] { "foo-company", "bar-company" });
+            result.Select(x => x.Name).Should().BeEquivalentTo(new[] { "Foo Company", "Bar Company" });
+        }
+
+        [TestMethod]
+        public async Task GetOrganizations_Paging()
+        {
+            await ConnectToSonarQube("6.2.0.0");
+
+            SetupRequest("api/organizations/search?p=1&ps=500", $@"
+{{
+  ""paging"": {{
+    ""pageIndex"": 1,
+    ""pageSize"": 500,
+    ""total"": 2
+  }},
+  ""organizations"": [
+    {string.Join(",\n", Enumerable.Range(1, 500).Select(i => $@"{{ ""key"": ""{i}"", ""name"": ""Name{i}"" }}"))}
+  ]
+}}");
+
+            SetupRequest("api/organizations/search?p=2&ps=500", $@"
+{{
+  ""paging"": {{
+    ""pageIndex"": 2,
+    ""pageSize"": 500,
+    ""total"": 2
+  }},
+  ""organizations"": [
+    {string.Join(",\n", Enumerable.Range(501, 500).Select(i => $@"{{ ""key"": ""{i}"", ""name"": ""Name{i}"" }}"))}
+  ]
+}}");
+
+            SetupRequest("api/organizations/search?p=3&ps=500", $@"
+{{
+  ""paging"": {{
+    ""pageIndex"": 3,
+    ""pageSize"": 500,
+    ""total"": 2
+  }},
+  ""organizations"": [
+    {string.Join(",\n", Enumerable.Range(1001, 10).Select(i => $@"{{ ""key"": ""{i}"", ""name"": ""Name{i}"" }}"))}
+  ]
+}}");
+
+            var result = await service.GetAllOrganizationsAsync(CancellationToken.None);
+
+            messageHandler.VerifyAll();
+
+            result.Should().HaveCount(1010);
+            result.Select(x => x.Key).Should().BeEquivalentTo(Enumerable.Range(1, 1010).Select(i => i.ToString()));
+            result.Select(x => x.Name).Should().BeEquivalentTo(Enumerable.Range(1, 1010).Select(i => $"Name{i}"));
+        }
+
+        [TestMethod]
+        public async Task GetOrganizations_NotFound()
+        {
+            await ConnectToSonarQube("6.2.0.0");
+
+            SetupRequest("api/organizations/search?p=1&ps=500", "", HttpStatusCode.NotFound);
+
+            Func<Task<IList<SonarQubeOrganization>>> func = async () => await service.GetAllOrganizationsAsync(CancellationToken.None);
+
+            func.Should().ThrowExactly<HttpRequestException>().And
+                .Message.Should().Be("Response status code does not indicate success: 404 (Not Found).");
+
+            messageHandler.VerifyAll();
+        }
+
+        [TestMethod]
+        public async Task GetOrganizations_Old_SonarQube()
+        {
+            await ConnectToSonarQube("6.1.0.0");
+
+            Func<Task> action = async () => await service.GetAllOrganizationsAsync(CancellationToken.None);
+
+            action.Should().ThrowExactly<InvalidOperationException>();
```

Comment:
Are we not expecting any specific message?

Reply:
The message is covered in the tests for RequestFactory, not sure if we should duplicate the message everywhere.

Reply:
@valhristov there too.